polkadot/node/core/approval-voting/src/lib.rs

```diff
@@ -1864,6 +1866,7 @@ async fn distribution_messages_for_activation<Sender: SubsystemSender<RuntimeApi
 	}
 
 	messages[0] = ApprovalDistributionMessage::NewBlocks(approval_meta);
+	messages.extend(approvals.into_iter());
```

Comment:
I'd also add a comment here to explain why we send all approvals after assignments.

Reply:
done.